crates/egui/src/containers/popup.rs

```diff
@@ -1,457 +1,517 @@
-//! Show popup windows, tooltips, context menus etc.
-
-use pass_state::PerWidgetTooltipState;
-
 use crate::{
-    pass_state, vec2, AboveOrBelow, Align, Align2, Area, AreaState, Context, Frame, Id,
-    InnerResponse, Key, LayerId, Layout, Order, Pos2, Rect, Response, Sense, Ui, UiKind, Vec2,
-    Widget, WidgetText,
+    Area, AreaState, Context, Frame, Id, InnerResponse, Key, LayerId, Layout, Order, Response,
+    Sense, Ui, UiKind,
 };
+use emath::{vec2, Align, Align4, Pos2, Rect};
+use std::iter::once;
 
-// ----------------------------------------------------------------------------
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PopupAnchor {
+    /// Show the popup at the given rect.
+    Rect(Rect),
```

Comment:
Will the popup always have the given size, even if its contents is smaller? What if it overflows?